.github/workflows/tests.yml

```diff
@@ -25,49 +25,34 @@ on:
         default: 'Manual trigger'
 
 jobs:
-  Tests:
-    runs-on: ubuntu-20.04
-    continue-on-error: ${{ matrix.ignore-failure }}
+  Python:
+    runs-on: ubuntu-24.04
     strategy:
       matrix:
           package-dir: ["site", "legacy"]
-          python-version: [3.9]
+          python-version: ["3.9"]
           db-service: [postgresql14]
           search-service: [opensearch2]
-          ignore-failure: [false]
-          include:
-          - package-dir: "migrator"
-            ignore-failure: true
-            python-version: 3.9
-            db-service: postgresql14
     env:
       DB: ${{ matrix.db-service }}
       SEARCH: ${{ matrix.search-service }}
       EXTRAS: tests
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
+
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
-        with:
-          node-version: ${{ matrix.node-version }}
-
-      - name: Run eslint test
-        run: ./run-js-linter.sh -i
-
-      - name: Generate dependencies
-        run: |
-          pip install "pipenv==2023.12.1"
+          cache: pip
+          cache-dependency-path: Pipfile.lock
 
       - name: Install dependencies
         run: |
           sudo apt-get update
           sudo apt-get -y install libkrb5-dev libvips-dev
+          pip install "pipenv==2023.12.1"
```

Comment:
Should we document why we are pining the `pipenv` version here?

Reply:
Yes, in the end actually this wasn't necessary, but we wanted to match the `pipenv` version that's on the Docker image that is deployed.

I'll add a comment